frontend/js/src/common/brainzplayer/NavidromePlayer.tsx

```diff
@@ -0,0 +1,436 @@
+import * as React from "react";
+import { get as _get, isString, throttle as _throttle } from "lodash";
+import { Link } from "react-router";
+import { faNavidrome } from "../icons/faNavidrome";
+import {
+  DataSourceProps,
+  DataSourceType,
+  DataSourceTypes,
+} from "./BrainzPlayer";
+import {
+  getArtistName,
+  getTrackName,
+  searchForNavidromeTrack,
+} from "../../utils/utils";
+import GlobalAppContext from "../../utils/GlobalAppContext";
+import { dataSourcesInfo } from "../../settings/brainzplayer/BrainzPlayerSettings";
+
+export type NavidromePlayerState = {
+  currentTrack?: NavidromeTrack;
+};
+
+export type NavidromePlayerProps = DataSourceProps;
+
+export default class NavidromePlayer
+  extends React.Component<NavidromePlayerProps, NavidromePlayerState>
+  implements DataSourceType {
+  static contextType = GlobalAppContext;
+  static hasPermissions = (navidromeUser?: NavidromeUser) => {
+    return Boolean(
+      navidromeUser?.md5_auth_token && navidromeUser?.instance_url
+    );
+  };
+
+  static isListenFromThisService(listen: Listen | JSPFTrack): boolean {
+    const musicService = _get(
+      listen,
+      "track_metadata.additional_info.music_service"
+    );
+    return (
+      isString(musicService) && musicService.toLowerCase().includes("navidrome")
+    );
+  }
+
+  public name = "navidrome";
```

Comment:
My bad! I previously asked you to remove `domainName`, but it is causing an error as it is expected by other parts of the code.

However for our use case we can set it to `false` rather than the string "navidrome.org", which wouldn't make sense in this context as it is used to construct listen metadata.


```suggestion
  public name = "navidrome";
  public domainName = false;
```